### DISCUSSION TITLE
CBDLP Preview image is assumed to have 3 channels, would it be correct for RGBA images?

### DISCUSSION THREAD

**Discussion by Megidd** [sn4k3/UVtools#217] [Q&A] [answered]
Following #216 I'm trying to debug why this file cannot be opened by UVtools due to a crash without any error log in the settings folder:

[teapot_2021-06-09 (3).zip](https://github.com/sn4k3/UVtools/files/6641732/teapot_2021-06-09.3.zip)

Debugging through the code shows why the crash happens. The new image declared to store image data is of size `100*184*3 = 55200` but the `pixel` counter goes up to `57281`:

![image](https://user-images.githubusercontent.com/17475482/121766594-54856e00-cb68-11eb-92bb-4ce93b119488.png)

### Cause?

Looking more, it's discovered that the assumed channels for the preview image are `3`, but the preview image has `4` channels of RGBA:

![image](https://user-images.githubusercontent.com/17475482/121766659-dd040e80-cb68-11eb-941a-23c39de1fc9e.png)

Could this be the cause of the crash?

**Comment by Megidd** [comment, marked as answer]
An [old library](https://github.com/Andoryuuta/photon) was being used to generate the attached CBDDLP file out of a collection of layer PNG images.

Further tests indicate that the old library has some sort of bug, not sure exactly how. So, finally, abandoning it in favor of a [newer repository](https://github.com/ezrec/uv3dp) suggested by @sn4k3 here: https://github.com/sn4k3/UVtools/discussions/216#discussioncomment-847314

**Reply by sn4k3** [reply]
Preview images must be 3 channels. Otherwise you will crash printer too. But is easy to discard the A channel on writing...
Pixel is greater than image size because your DATA is larger than the image buffer and i'm not checking for safeness. I could put a if condition to avoid such problems and trigger an error...

On the old library they are using RGB too, bug must be on other line
